lib/variant-model.js

```diff
@@ -45,17 +47,42 @@ _.extend(Variant.prototype, {
     return this;
   },
 
-  respondWithFile: function(fileName) {
+  respondWithFile: function(options) {
+    options = _.extend({
+      variant_id: this.id(),
+      path: this._route.path(),
+    }, options);
+
     return this.respondWith(function(request, reply) {
-      var self = this;
-      fileName = fileName.replace(/\{([^\}]+)\}/g, function(match, token) {
-        var val = request.params[token];
-        if (!val) {
-          val = self.state(token);
+      var smocks = require('./index');
+
+      options = _.extend({
```

Comment:
shouldn't be populating `options` twice.  just incorporate https://github.com/jhudson8/smocks/pull/8/files#diff-9ca30b032382b748e0cc599a1cafae48R51 into here.  Remember to create a reference to `this` outside of this function block and use `self` instead.